include/internal/catch_result_builder.hpp

```diff
@@ -130,27 +122,31 @@ namespace Catch {
         }
 
         data.message = m_stream.oss.str();
-        data.reconstructedExpression = reconstructExpression();
-        if( m_exprComponents.testFalse ) {
-            if( m_exprComponents.op == "" )
-                data.reconstructedExpression = "!" + data.reconstructedExpression;
-            else
-                data.reconstructedExpression = "!(" + data.reconstructedExpression + ")";
+        if( data.resultType == ResultWas::ExpressionFailed ) {
```

Comment:
I think there are cases where Catch can be made to print expressions even when they don't fail (e.g. `--success` and possibly the JUnit reporter). Therefore always suppressing expression building for non-failed tests might be incorrect.